medusa/storage/s3_base_storage.py

```diff
@@ -328,18 +346,29 @@ async def _upload_blob(self, src: str, dest: str) -> ManifestObject:
             )
         )
 
-        self.s3_client.upload_file(
-            Filename=src,
-            Bucket=self.bucket_name,
-            Key=object_key,
-            Config=self.transfer_config,
-            ExtraArgs=kms_args,
-        )
-
-        blob = await self._stat_blob(object_key)
-        mo = ManifestObject(blob.name, blob.size, blob.hash)
+        upload_conf = {
+            'Filename': src,
+            'Bucket': self.bucket_name,
+            'Key': object_key,
+            'Config': self.transfer_config,
+            'ExtraArgs': kms_args,
+        }
+        # we are going to combine asyncio with boto's threading
+        # we do this by submitting the upload into an executor
+        loop = asyncio.get_event_loop()
+        future = loop.run_in_executor(self.executor, self.__upload_file, upload_conf)
+        # and then ask asyncio to yield until it completes
+        mo = await future
```

Comment:
**Question:** I fail to understand how that parallelizes uploads. Here you're uploading a single file and then wait for the future (so the upload) to complete, until you move to the next one. Or am I missing something?

Reply:
The way I suspect this works is that the `await` keyword _yields_ the execution of the current coroutine. 

The `_upload_object` is async already, which means several of them happen concurrently. The issue was they didn't because the call to s3_client.upload() was blocking.

With the change here, we hide the blocking call to client into a future, and the yield until it completes.